test/Integration/conv-to-matmul.mlir

```diff
@@ -195,5 +204,10 @@ func.func @entry() {
   // TRANSFORM-SAME:   ( 0, 28, 56, 84, 112, 140, 168, 196 ) ) ) )
   //
   vector.print %v2 : vector<1x2x2x8xf32>
+
+  bufferization.dealloc_tensor %img_seed : tensor<?xf32>
```

Comment:
is this not inserted by bufferization?

Reply:
Nope, there are memory leaks without this explicit deallocation.

Reply:
Yeah, the IR itself allocates them, so it should also deallocate.

Reply:
I did not see any leaks here for example: https://github.com/plaidml/tpp-mlir/commit/464116e4517d06db4d1a6bae0c83f664c4ebaa3d

Reply:
Although, now that I double-checked it, the leaks are present only when the default pipeline is used. The older `RUN` statements execute correctly. I'll need to have a look where this difference is coming from.

Reply:
The generated LLVM IR (manual pipeline vs tpp-run) looks to be mostly the same - identical in the number of allocs and frees. I think that our `mlir-cpu-runner` just doesn't include any sanitizers (I tried building LLVM with sanitizers but ran into some errors) whereas they are enabled for `tpp-run` which then reports the memory leaks.

@chelini one-shot or `BufferDeallocationPass` cannot handle cases when an explicitly allocated buffer is returned from a function. In the example you linked it is fine because the alloc is within the main function.

Reply:
You can build `mlir-cpu-runner` with santizers (may need some pushing, may have temporary upstream bugs on the commit we choose, etc), but you have to build the entire LLVM+MLIR. 

It's doable, but using `tpp-run` directly is much easier and more controllable.